unstructured/ingest/connector/elasticsearch.py

```diff
@@ -299,3 +306,96 @@ def get_ingest_docs(self):
             )
             for batched_ids in id_batches
         ]
+
+
+@dataclass
+class ElasticsearchWriteConfig(WriteConfig):
+    batch_size_bytes: int
+    num_processes: int
+
+
+@dataclass
+class ElasticsearchDestinationConnector(BaseDestinationConnector):
+    write_config: ElasticsearchWriteConfig
+    connector_config: SimpleElasticsearchConfig
+    _client: t.Optional["Elasticsearch"] = field(init=False, default=None)
+
+    @DestinationConnectionError.wrap
+    @requires_dependencies(["elasticsearch"], extras="elasticsearch")
+    def generate_client(self) -> "Elasticsearch":
+        from elasticsearch import Elasticsearch
+
+        return Elasticsearch(
+            **self.connector_config.access_config.to_dict(apply_name_overload=False)
+        )
+
+    @property
+    def client(self):
+        if self._client is None:
+            self._client = self.generate_client()
+        return self._client
+
+    def initialize(self):
+        _ = self.client
+
+    @DestinationConnectionError.wrap
+    def check_connection(self):
+        try:
+            assert self.client.ping()
+        except Exception as e:
+            logger.error(f"failed to validate connection: {e}", exc_info=True)
+            raise DestinationConnectionError(f"failed to validate connection: {e}")
+
+    def get_document_size(doc):
```

Comment:
Does this ever run? I don't see it called anywhere.

Reply:
Removed with [remove unused method, update schema](https://github.com/Unstructured-IO/unstructured/pull/2152/commits/e2fe2c56102d27fb10427ae9ba0e9bbb53658da5)